### PR TITLE
remove Mkdirtemp func to reduce util pkg for testing

### PR DESCRIPTION
**Pull request by charles-chenzz**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
this pr follow what #117385 post and reduce util pkg for testing(call less than 10), I also replace deprecated ioutil with os
#### Which issue(s) this PR fixes:

parts of #117385 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by charles-chenzz** [comment]
this is part of the issues and I still reseaching where we can refactoring the util pkg for testing. 
1. In ``staging/src/k8s.io/cri-api/pkg/apis/testing/util.go``, there are two helper func which call by ``fake_runtime_service.go`` just one times. but I don't know if we're comfortable about remove it.
https://github.com/kubernetes/kubernetes/blob/53cccbe4f98b9c6086af88954f5c9a77d8e66fc5/staging/src/k8s.io/cri-api/pkg/apis/testing/utils.go#L26-L29
https://github.com/kubernetes/kubernetes/blob/53cccbe4f98b9c6086af88954f5c9a77d8e66fc5/staging/src/k8s.io/cri-api/pkg/apis/testing/utils.go#L32-L34

2. I didn't remove the ``staging/src/k8s.io/client-go/util/testing/tmpdir.go`` yet, cus I want @thockin to check if I've missing something.

/assign @thockin

File: staging/src/k8s.io/cli-runtime/pkg/resource/builder_test.go [hunk 1]
```diff
@@ -704,7 +703,7 @@ func TestErrorFilePatternBuilder(t *testing.T) {
 }
 
 func setupKustomizeDirectory() (string, error) {
-	path, err := ioutil.TempDir("/tmp", "")
+	path, err := os.MkdirTemp("/tmp", "")
```
Comment (thockin) [inline staging/src/k8s.io/cli-runtime/pkg/resource/builder_test.go:706]: I think ALL of these calls to `os.MkdirTemp` can pass "" as the first arg, no?

**Review by thockin — COMMENTED** [review]
Does this allow us to mark those functions as deprecated?  We should do that in the same PR.

File: staging/src/k8s.io/cli-runtime/pkg/resource/builder_test.go (hunk 1 above)
Comment (charles-chenzz) [reply staging/src/k8s.io/cli-runtime/pkg/resource/builder_test.go]: yes, it can pass "" as the first arg,  and I updated it. PTAL @thockin

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**